rholang/src/test/scala/coop/rchain/rholang/interpreter/matcher/StreamTSpec.scala

```diff
@@ -106,11 +109,44 @@ class StreamTLawsSpec extends CatsSuite {
       )
     )
 
-  implicit def eqEff[A: Eq]: Eq[Effect[A]]       = Eq.by(x => (x.value.value))
+  implicit val arbFunctionKStream: Arbitrary[Stream ~> Stream] =
+    Arbitrary(
+      Gen.oneOf(
+        new (Stream ~> Stream) {
+          def apply[A](fa: Stream[A]): Stream[A] = Stream.Empty
+        },
+        FunctionK.id[Stream]
+      ))
+
+  implicit def eqEff[A: Eq]: Eq[Effect[A]]       = Eq.by(x => x.value.value.value())
   implicit def eqFA[A: Eq]: Eq[StreamTEffect[A]] = Eq.by(StreamT.run[Effect, A])
 
-  checkAll("StreamT.MonadLaws", MonadTests[StreamTEffect].monad[Int, Int, String])
-  checkAll("StreamT.MonoidKLaws", MonoidKTests[StreamTEffect].monoidK[Int])
-  checkAll("StreamT.MonadTransLaws", MonadTransTests[StreamT].monadTrans[Effect, Int, String])
+  checkAll(
+    "StreamT.MonadLaws",
+    MonadTests[StreamTEffect].monad[Int, Int, String]
+  )
+  checkAll(
+    "StreamT.MonoidKLaws",
+    MonoidKTests[StreamTEffect].monoidK[Int]
+  )
+  checkAll(
+    "StreamT.MonadTransLaws",
+    MonadTransTests[StreamT].monadTrans[Effect, Int, String]
+  )
+  checkAll(
+    "StreamT.MonadLayerControlLaws",
+    MonadLayerControlTests[StreamTEffect, Effect, Stream].monadLayerControl[Int, String]
+  )
+  checkAll(
+    "StreamT.MonadErrorLaws",
+    MonadErrorTests[StreamTEffect, String].monadError[Int, Int, String]
+  )
 
 }
+
+trait LowPriorityDerivations {
+
+  implicit def arbFunctionK[K[_]]: Arbitrary[K ~> K] =
```

Comment:
why do you generate arbitrary for `K~>K`? isn't there a single instance of this transformation?

Reply:
There is, but not having this caused implicit resolution to fail. From what I understand, this is because of several layers of transformers used in this test. Analogous tests in cats use just one layer, and can get away with just a single instance similar to `arbFunctionKStream` I've got above.